### ISSUE TITLE
json 2.7.3 has broken my test environment

### ISSUE THREAD

**Issue by david22swan** [ruby/json#646]
Hey, sorry to bother but have had issues with the newest release of the Gem with the below error blocking my test development:

```
An error occurred while loading ./spec/acceptance/set_config_spec.rb.
On host `localhost:52362'
Failure/Error: require 'json'
ArgumentError:
  wrong number of arguments (given 1, expected 0)

# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/common.rb:82:in `initialize'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/common.rb:82:in `new'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/common.rb:82:in `create_fast_state'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/common.rb:75:in `generator='
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/ext.rb:11:in `<module:Ext>'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/ext.rb:6:in `<module:JSON>'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json/ext.rb:3:in `<top (required)>'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json.rb:579:in `<module:JSON>'
# ./.bundle/ruby/3.2.0/gems/json_pure-2.6.3/lib/json.rb:575:in `<top (required)>'
# ./spec/spec_helper_acceptance.rb:8:in `<top (required)>'
# ./spec/acceptance/set_config_spec.rb:1:in `<top (required)>'
```

Was able to resolve this by pinning the json gem itself to 2.7.2 so can confirm the issue is with the new release, and thought I should shout out incase anyone else has gotten hit with it.
Unsure of the actual cause of the error, or if it has something to do with us using json_pure and json being pulled in from another dependency.

**Comment by byroot** [comment]
Ah yes, `json_pure` and `json` need to be at the same version if both are co-existing.

**Comment by david22swan** [comment]
Was able to fix the issue with them different though.
Fixed it by pinning json to 2.7.2 while json_pure was still pinned to 2.6.3?

**Comment by byroot** [comment]
Yes, some version mix can sometime work, but they are only guaranteed to work if the version matches.

Unfortunately the `json_pure 2.7.3` release is delayed for technical reasons, but it should be released some time next week:  https://github.com/ruby/json/issues/640

So I'll close, as the failure is unfortunately expected.

**Comment by casperisfine** [comment]
Someone else reported a similar issue, and after investigation it's caused by a rubygems bug: https://github.com/rubygems/rubygems/pull/6490

Upgrading rubygems solves it `gem update --system`

**Comment by casperisfine** [comment]
Actually, ignore the above, it's the same problem but a different trigger.

I'm working on a workaround in https://github.com/ruby/json/pull/648 so that hopefully `2.7.4` will work OK even if loaded with a different version of `json_pure`, but long term I'm hoping to find a better fix.

**Comment by byroot** [comment]
I released `2.7.4` with a workaround for such situation, but I'll try to find a better solution to allow independent use of `json_pure` and `json` for 2.8.0.
